pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -375,6 +375,20 @@ type derivedInput struct {
 	Values      map[string]any
 }
 
+func setInputValue(input derivedInput, field string, value any) {
+	entry := input.Values
+	keys := strings.Split(field, ".")
+	for i := 0; i < len(keys)-1; i++ {
+		val, ok := entry[keys[i]]
+		if !ok {
+			val = map[string]any{}
+			entry[keys[i]] = val
+		}
+		entry = val.(map[string]any)
```

Comment:
I think the map we are setting is shared between sidecar injection as well. Is mutating this going to be safe? At the very least seems like we might get a race from read+write at the same time.

It may make sense to re-use the code in overwriteClusterInfo which does this after rendering instead of as input?

We could also just add a Cluster param to TemplateInput and use that.

Reply:
If using overwriteClusterInfo is hard due to Unstructured, I may have a change coming that makes it easier... let me get back to you soon (if its a blocker).

Reply:
https://github.com/istio/istio/pull/43541 was the change. Not sure it helps much

Reply:
Added ClusterID to TemplateInput